src/hw/hw_mngr.cpp

```diff
@@ -13,10 +13,148 @@
 
 #include <charconv>
 #include <filesystem>
+#include <iostream>
+#include <fstream>
+#include <sstream>
 
 namespace fs = std::filesystem;
 using namespace phosphor::logging;
 
+namespace fans
+{
+constexpr int maxErrorAttempts = 20;
+constexpr const char* emptyString = "";
+constexpr const char* sysHwmonPath = "/sys/class/hwmon/";
+constexpr const char* sysHwmonFile = "name";
+constexpr const char* sysHwmonContent = "aspeed_pwm_tacho";
+
+struct FanInfo
+{
+    const uint32_t inletRangeMin{0};
+    const uint32_t inletRangeMax{0};
+    const uint32_t outletRangeMin{0};
+    const uint32_t outletRangeMax{0};
+    const char* partNumber{nullptr};
+    const char* prettyName{nullptr};
+
+    bool operator== (const FanInfo& o) const {
+        return (this == &o) ? true : false;
+    }
+    bool operator!= (const FanInfo& o) const {
+        return (this != &o) ? true : false;
+    }
+};
+
+std::vector<FanInfo> detectionFanTable = {
+    {19000, 24000, 16200, 19800, "ASMFAN783104A", "Standard performance fan"},
+    {26550, 34000, 22950, 30800, "ASMFAN781104A1", "High performance fan"}
+};
+
+static const FanInfo mixedFan;
+static const FanInfo invalidFan;
+
+const FanInfo& getFanInfoByRPM(uint32_t inletRpm, uint32_t outletRpm)
+{
+    bool bFoundMixedFan = false;
+    for (const auto& item : detectionFanTable)
+    {
+        bool inletRpmMatched =
+            (inletRpm >= item.inletRangeMin && inletRpm <= item.inletRangeMax);
+        bool outletRpmMatched = (outletRpm >= item.outletRangeMin &&
+                                 outletRpm <= item.outletRangeMax);
+
+        if (inletRpmMatched && outletRpmMatched)
+            return item;
+
+        if (inletRpmMatched || outletRpmMatched)
+            bFoundMixedFan = true;
```

Comment:
What is `mixed` in your understanding? I think, mixed fans it is when you have some normal and some HiPerf fans in one chassis (which shouldn't be in real world).
When FanA detected as "normal" and FanB detected as "Hi Perf" it is broken (or at least "invalid") fan. They physically can't be mixed as they are solid single part, assembled by vendor and can't be reassembled.

Reply:
fixed